python/cudf/cudf/core/reshape.py

```diff
@@ -36,6 +38,12 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
     if len(objs) == 1:
         return objs[0]
 
+    # convert any RangeIndex
+    objs = [
+        as_index(obj._values) if isinstance(obj, RangeIndex) else obj
+        for obj in objs
+    ]
```

Comment:
Why do we need to materialize the RangeIndex here instead of in `Index._concat`?

Reply:
Because there's a check below that all `objs` are of the same `type`. We should keep that check for `Index` objects, but after converting all `RangeIndex` to a materialized index.

Reply:
I guess this is to allow concatenating a RangeIndex with a GenericIndex for example. Makes sense.